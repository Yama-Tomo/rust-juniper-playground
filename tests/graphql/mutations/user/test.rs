#[path = "../../../test_utils/mod.rs"]
mod test_utils;

use insta::*;
use serde_json::json;
use test_context::test_context;

use test_utils::*;

#[test_context(TestContext)]
#[actix_rt::test]
async fn ユーザを追加することができること(ctx: &TestContext) {
    let (app, _) = create_service().await;
    let req_body = json!({
       "query": "mutation Test {
          addUser(input:{name: \"John\"}) {
            ... on User {
              id
              name
            }
            ... on ValidationErrors {
              errors {
                field
                message
              }
            }
          }
        }"
    });

    let res = graphql_req(&app, req_body).await;
    assert_json_snapshot!(res);

    let req_body = json!({
       "query": "query ConfirmAddUser($id: Int!) {
          user(id: $id) {
            id
            name
          }
        }",
       "variables": { "id": res["data"]["addUser"]["id"] }
    });

    assert_json_snapshot!(graphql_req(&app, req_body).await);
}

#[test_context(TestContext)]
#[actix_rt::test]
async fn 空のユーザ名の場合はバリデーションエラーをレスポンスすること(
    ctx: &TestContext,
) {
    let (app, _) = create_service().await;
    let req_body = json!({
       "query": "mutation Test {
          addUser(input:{name: \"\"}) {
            ... on User {
              id
              name
            }
            ... on ValidationErrors {
              errors {
                field
                message
              }
            }
          }
        }"
    });

    assert_json_snapshot!(graphql_req(&app, req_body).await);
}
