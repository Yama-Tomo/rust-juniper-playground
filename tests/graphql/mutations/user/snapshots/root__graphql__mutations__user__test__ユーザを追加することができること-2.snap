---
source: tests/graphql/mutations/user/test.rs
assertion_line: 44
expression: "graphql_req(&app, req_body).await"

---
{
  "data": {
    "user": {
      "id": 5,
      "name": "John"
    }
  }
}
